### PR TITLE
Global sub id v2plus

### PR DESCRIPTION
**Pull request by jinhoonbang**

### CODE REVIEW

File: contracts/src/v0.8/dev/vrf/SubscriptionAPI.sol [hunk 1]
```diff
@@ -338,4 +346,8 @@ abstract contract SubscriptionAPI is ConfirmedOwner, ReentrancyGuard, ERC677Rece
     }
     _;
   }
+
+  function getConsumerKey(address consumer, uint256 subId) internal pure returns (uint256) {
```
Comment (jinhoonbang) [inline contracts/src/v0.8/dev/vrf/SubscriptionAPI.sol:350]: can change this to return byte32

**Review by makramkd — COMMENTED** [review]
Looks good overall, small amount of concern passing around the version but nothing too serious.

File: contracts/src/v0.8/dev/vrf/SubscriptionAPI.sol [hunk 2]
```diff
@@ -338,4 +346,8 @@ abstract contract SubscriptionAPI is ConfirmedOwner, ReentrancyGuard, ERC677Rece
     }
     _;
   }
+
+  function getConsumerKey(address consumer, uint256 subId) internal pure returns (bytes32) {
```
[the remark this answers is not in the document]
Comment (jinhoonbang) [reply contracts/src/v0.8/dev/vrf/SubscriptionAPI.sol:350]: Good comment. I didn't realize this wasn't tested. I am actually seeing higher gas with this change. 

Before this change:
requestRandomWords
`avg: 33996`
`median: 32958`
`max: `37758`

after: 
requestRandomWords
`avg: 34347`
`median: 33309`
`max: 38109`

Will revert this change

File: core/chains/evm/txmgr/txmgr_test.go [hunk 3]
```diff
@@ -247,6 +247,8 @@ func TestTxm_CreateTransaction(t *testing.T) {
 		pgtest.MustExec(t, db, `DELETE FROM eth_txes`)
 		testDefaultSubID := uint64(2)
 		testDefaultMaxLink := "1000000000000000000"
+		// max uint256 is 1.1579209e+77
+		testDefaultGlobalSubID := new(big.Int).Exp(big.NewInt(1), big.NewInt(77), nil).String()
```
[the remark this answers is not in the document]
Comment (jinhoonbang) [reply core/chains/evm/txmgr/txmgr_test.go:251]: oops. will use the second suggestion 👍

File: core/services/vrf/v2/coordinator_v2x_interface.go [hunk 4]
```diff
@@ -237,9 +237,9 @@ func (r *RandomWordsRequested) NumWords() uint32 {
 	return r.V2Plus.NumWords
 }
 
-func (r *RandomWordsRequested) SubID() uint64 {
+func (r *RandomWordsRequested) SubID() *big.Int {
```
[the remark this answers is not in the document]
Comment (jinhoonbang) [reply core/services/vrf/v2/coordinator_v2x_interface.go:240]: Yes I think I prefer the suggested pattern as well. will refactor

File: contracts/src/v0.8/dev/vrf/VRFCoordinatorV2Plus.sol [hunk 5]
```diff
@@ -325,7 +325,7 @@ contract VRFCoordinatorV2Plus is VRF, SubscriptionAPI {
   function computeRequestId(
     bytes32 keyHash,
     address sender,
-    uint64 subId,
+    uint256 subId,
```
Comment (makramkd) [inline contracts/src/v0.8/dev/vrf/VRFCoordinatorV2Plus.sol:328]: Ah I think this would change the preseed for sure. That would explain the proof change.

**Review by makramkd — APPROVED** [review]

**Pull request merged**